core/src/main/java/com/linecorp/armeria/client/endpoint/DynamicEndpointGroup.java

```diff
@@ -260,15 +259,18 @@ protected final void setEndpoints(Iterable<Endpoint> endpoints) {
             return;
         }
 
-        endpointsLock.lock();
+        lock();
```

Comment:
Question) Would it be worth acquiring a lock before L258 so that `hasChanges` doesn't return early unintentionally?

Reply:
Good suggestion. Fixed.